alert-database/src/main/java/com/synopsys/integration/alert/database/api/DefaultAuthorizationUtility.java

```diff
@@ -0,0 +1,137 @@
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import com.synopsys.integration.alert.common.enumeration.AccessOperation;
+import com.synopsys.integration.alert.common.persistence.accessor.AuthorizationUtil;
+import com.synopsys.integration.alert.common.persistence.model.PermissionMatrixModel;
+import com.synopsys.integration.alert.common.persistence.model.UserRoleModel;
+import com.synopsys.integration.alert.database.authorization.AccessOperationEntity;
+import com.synopsys.integration.alert.database.authorization.AccessOperationRepository;
+import com.synopsys.integration.alert.database.authorization.PermissionKeyEntity;
+import com.synopsys.integration.alert.database.authorization.PermissionKeyRepository;
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRelation;
+import com.synopsys.integration.alert.database.authorization.PermissionMatrixRepository;
+import com.synopsys.integration.alert.database.user.RoleEntity;
+import com.synopsys.integration.alert.database.user.RoleRepository;
+import com.synopsys.integration.alert.database.user.UserRoleRelation;
+import com.synopsys.integration.alert.database.user.UserRoleRepository;
+
+@Component
+@Transactional
+public class DefaultAuthorizationUtility implements AuthorizationUtil {
+    private final RoleRepository roleRepository;
+    private final UserRoleRepository userRoleRepository;
+    private final PermissionMatrixRepository permissionMatrixRepository;
+    private final AccessOperationRepository accessOperationRepository;
+    private final PermissionKeyRepository permissionKeyRepository;
+
+    @Autowired
+    public DefaultAuthorizationUtility(final RoleRepository roleRepository, final UserRoleRepository userRoleRepository, final PermissionMatrixRepository permissionMatrixRepository, final AccessOperationRepository accessOperationRepository,
+        final PermissionKeyRepository permissionKeyRepository) {
+        this.roleRepository = roleRepository;
+        this.userRoleRepository = userRoleRepository;
+        this.permissionMatrixRepository = permissionMatrixRepository;
+        this.accessOperationRepository = accessOperationRepository;
+        this.permissionKeyRepository = permissionKeyRepository;
+    }
+
+    @Override
+    public Set<UserRoleModel> createRoleModels(final Collection<Long> roleIds) {
+        final Set<UserRoleModel> userRoles = new LinkedHashSet<>();
+        for (final Long roleId : roleIds) {
+            final Optional<String> roleName = getRoleName(roleId);
+            roleName.ifPresent(role -> userRoles.add(UserRoleModel.of(role, readPermissionsForRole(roleId))));
+        }
+        return userRoles;
+    }
+
+    @Override
+    public void updateUserRoles(final Long userId, final Collection<UserRoleModel> roles) {
+        if (null != userId) {
+            userRoleRepository.deleteAllByUserId(userId);
+        }
+
+        if (null != roles) {
```

Comment:
Is there a specific reason we do 2 different operations depending on the data passed us? This could cause some confusing functionality. You accidentally only pass a UserId and a null roles and now we've randomly deleted all userRoles in an update method. I would prefer we check for null in a single if statement then do all our steps.

Reply:
Need to make a change there to make it better.